Functions/xASL_io_ReadDataPar.m

```diff
@@ -0,0 +1,250 @@
+%               only added value to normal json-read is that it detects invalid entries (numbers in strings, and 
+%               weird arrays), converts them correctly and reports this incorrect entries so that they can be manually 
+%               fixed. Also, if an .m file is provided, it converts and saves it to a JSON file (doesn't overwrite)
+%               and reports that you should stop using .m files.
+%
+% -----------------------------------------------------------------------------------------------------------------------------------------------------
+%
+% EXAMPLE:      xASL_io_ReadDataPar('DataParFile.json')
+%
+% EXAMPLE 2:    xASL_io_ReadDataPar('DataParFile.m')
+%
+% EXAMPLE 2:    JSON FILE
+%
+% {
+% 	"x": [{
+% 			"name": 				"ExampleDataSet",
+% 			"subject_regexp": 		"^Sub-\\d{3}$",
+% 			"M0": 					"separate_scan",
+% 			"BackgroundSuppressionNumberPulses": 2,
+% 			"readout_dim": 			"2D",
+% 			"QUALITY": 				false,
+% 			"Vendor": 				"Philips",
+% 		}]
+% }
+%
+% Don't forget to escape the backslashes!
+%
+% EXAMPLE 3: To define cell arrays, use the official JSON array notation:
+%
+%   ...
+%   "S":
+%	{"Atlases": ["TotalGM","DeepWM","Hammers","HOcort_CONN","HOsub_CONN","Mindboggle_OASIS_DKT31_CMA"]}
+%   ...
+%
+% -----------------------------------------------------------------------------------------------------------------------------------------------------
+% __________________________________
+% Copyright 2015-2021 ExploreASL
+
+%% Input Check
+if nargin < 1 || isempty(DataParFile) || ~exist(DataParFile, 'file')
+    error('DataParFile does not exist...');
+end
+
+% Input has to be a character array
+DataParFile = char(DataParFile);
+[Fpath, Ffile, Fext] = fileparts(DataParFile);
+
+%% Read an m-file
+if strcmp(Fext, '.m')
+	% Read the m-file by executing it
+	CurrentFolder = pwd;
+	if ~isempty(Fpath)
+		cd(Fpath);
+	end
+
+	FunctionHandle = str2func(Ffile);
+	x = FunctionHandle();
+
+	cd(CurrentFolder);
+	
+	PathJSON = fullfile(Fpath, [Ffile '.json']);
+	
+	% Escape characters that are illegal in JSON
+	% Note that most characters (like backslashes) are correctly escaped when saving the JSON in spm_jsonwrite
+	FieldsX = fields(x);
+	for iField=1:length(FieldsX)
+		tString = x.(FieldsX{iField});
+		
+		if ~isstruct(tString)
+			% fix Boolean conversion issue JSON
+			if islogical(tString) && tString
+				tString = 1;
+			elseif islogical(tString) && ~tString
+				tString = 0;
+			end
+			
+			x.(FieldsX{iField}) = tString;
+		end
+	end
+	% Write the JSON if it doesn't exist
+	if exist(PathJSON, 'file')
+		fprintf('Warning: m-files are deprecated, but cannot convert to json-file because a file with json extension exists in the same directory');
+	else
+		fprintf('Warning: m-files are deprecated, converted and saved to json-file. Please delete the original m-file');
+		spm_jsonwrite(PathJSON, x);
+	end
+
+elseif strcmp(Fext, '.json')
```

Comment:
Should we do `strcmpi` to be forgiving if someone gets the capitals wrong (e.g. uses DataPar.Json and stalks us because the pipeline doesn't work ;)

Reply:
Yes. Done everywhere.